### PR TITLE
fix(datepicker): generate weekday labels in correct order

### PR DESCRIPTION
**Pull request by sorin-davidoi**
`timeZone` defaults to the user's timezone, which means that the week names might be returned in an unexpected order:

```
> TZ=Europe/Copenhagen node --eval "console.log([...Array(7).keys()].map((day) => Intl.DateTimeFormat('en', { weekday: 'narrow' }).format(new Date(Date.UTC(2021, 5, day - 1)))))"
[
  'S', 'M', 'T',
  'W', 'T', 'F',
  'S'
]

> TZ=America/New_York  node --eval "console.log([...Array(7).keys()].map((day) => Intl.DateTimeFormat('en', { weekday: 'narrow' }).format(new Date(Date.UTC(2021, 5, day - 1)))))"
[
  'S', 'S', 'M',
  'T', 'W', 'T',
  'F'
]
```

This would have been caught if the unit tests were run in different timezones (presumably they are run with `TZ=UTC` in CI?), as they are currently either passing or failing, depending on the timezone they are run in:

```
> TZ=UTC npm run test
...
TOTAL: 1579 SUCCESS
...

> TZ=America/New_York npm run test
...
TOTAL: 11 FAILED, 1568 SUCCESS
...
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#timezone

Closes https://github.com/ng-bootstrap/ng-bootstrap/issues/4729.

Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) and [DEVELOPER.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/DEVELOPER.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

**Comment by sorin-davidoi** [comment]
> This would have been caught if the unit tests were run in different timezones (presumably they are run with TZ=UTC in CI?), as they are currently either passing or failing, depending on the timezone they are run in.

Depending on how much resources can be used in CI, there are a few things we could do:
 - Run the tests (or a subset of them) in every timezone (they can be fetched on most Linux systems via `timedatectl list-timezones`)
 - Pick a somewhat representative list of timezones (one hour increments from UTC-12 to UTC+14, plus a few edge cases like Asia/Kathmandu?) and run the tests (or a subset of them) in those

With the usage of a [matrix strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy) it should be possible to keep the running times reasonable. If there's agreement on what to do I could take a stab at changing the CI to increase the test coverage.

**Comment by bastienmoulia** [comment]
Hi Sorin, your are right about the test, we should mock some other timezone.

Also can you add your fix in the other place where `Intl.DateTimeFormat` is used?

**Comment by samvloeberghs** [comment]
when is 17.1.0 milestone expected?
This is not a small bug, any chance we can expect this sooner in a patch release? Thank you for considering..

File: src/datepicker/datepicker-i18n.ts [hunk 1]
```diff
@@ -105,18 +105,20 @@ export class NgbDatepickerI18nDefault extends NgbDatepickerI18n {
 	private _locale = inject(LOCALE_ID);
 
 	private _monthsShort = [...Array(12).keys()].map((month) =>
-		Intl.DateTimeFormat(this._locale, { month: 'short' }).format(new Date(2000, month)),
+		Intl.DateTimeFormat(this._locale, { month: 'short', timeZone: 'UTC' }).format(new Date(2000, month, 15)),
```
Comment (leifjones) [inline src/datepicker/datepicker-i18n.ts:108]: What is the significance of this date? (Similar question for the 2021 date below.)

**Comment by maxokorokov** [comment]
Thanks

**Comment by samvloeberghs** [comment]
Thx for the release @maxokorokov.
Very appreciative of your time 🙏🏼
